src/inventory_ui.cpp

```diff
@@ -840,6 +840,12 @@ item_location game::inv_map_splice( item_filter filter, const std::string &title
     return inv_map_splice( filter, filter, filter, title, radius );
 }
 
+void clear_cats( std::vector<item_category *> &cats ) {
```

Comment:
Instead of this, why not put all the temporary categories into `unique_ptr`s inside a vector and let the compiler decide when to delete them?